### PR TITLE
Force unique ID for all AMP accordions

### LINKED ISSUE #18516: AMP accordion and accessibility on Lighthouse audit
amp-accordion is failing Lighthouse's accessibility test. The message from the test is the following:

<id> attributes on the page are not unique 
The value of an id attribute must be unique to prevent other instances from being overlooked by assistive technologies. 

How to reproduce:
run Lighthouse audit against any AMP that has an accordion.

### PR DESCRIPTION
**Pull request by nainar**
Fixes #18516.

Generates a random ID for all accordions that don't have a user specified ID to ensure uniqueness.

Note that this doesn't fix the case where users use the same ID for multiple accordions. That burden still lies on the web developers.

### CODE REVIEW

File: extensions/amp-accordion/0.1/amp-accordion.js [hunk 1]
```diff
@@ -64,6 +64,11 @@ class AmpAccordion extends AMP.BaseElement {
     /** @private {?../../../src/service/action-impl.ActionService} */
     this.action_ = null;
 
+    /** @private {string} */
```
Comment (nainar) [inline extensions/amp-accordion/0.1/amp-accordion.js]: Storing one generated suffix per accordion

File: extensions/amp-accordion/0.1/test/test-amp-accordion.js [hunk 2]
```diff
@@ -80,6 +80,20 @@ describes.realWin('amp-accordion', {
         });
       });
 
+  it('multiple accordions should not have the same IDs on content',
```
Comment (nainar) [inline extensions/amp-accordion/0.1/test/test-amp-accordion.js]: Added a unit test

**Comment by cathyxz** [comment]
Yup LGTM. Thank you!

**Comment by AndrewKGuan** [comment]
Hi,
has this been live yet?  I'm still seeing the [id] failing Lighthouse accessibility tests.

File: extensions/amp-accordion/0.1/amp-accordion.js [hunk 3]
```diff
@@ -64,6 +64,11 @@ class AmpAccordion extends AMP.BaseElement {
     /** @private {?../../../src/service/action-impl.ActionService} */
     this.action_ = null;
 
+    /** @private {number|string} */
+    this.suffix_ = element.id ? element.id :
+      Math.floor(Math.random() * Math.floor(100));
```
Comment (jridgewell) [inline extensions/amp-accordion/0.1/amp-accordion.js]: `Math.floor(100)` is just `100`.

File: extensions/amp-accordion/0.1/amp-accordion.js [hunk 4]
```diff
@@ -98,7 +103,11 @@ class AmpAccordion extends AMP.BaseElement {
       content.classList.add('i-amphtml-accordion-content');
       let contentId = content.getAttribute('id');
       if (!contentId) {
-        contentId = this.element.id + '_AMP_content_' + index;
+        // To ensure that we pass Accessibility audits -
+        // we need to make sure that each accordion has a unique ID.
+        // In case the accordion doesn't have an ID we use a
+        // random number to ensure uniqueness.
+        contentId = this.suffix_ + '_AMP_content_' + index;
```
Comment (jridgewell) [inline extensions/amp-accordion/0.1/amp-accordion.js]: Prefix. 😉

**Comment by AndrewKGuan** [comment]
Hi,
when would this go live?  Thanks!

**Comment by cathyxz** [comment]
@AndrewKGuan usually approx 2 weeks after merge. The fail safe way would be to check the [releases page](https://github.com/ampproject/amphtml/releases) for the PR number #18534, which is in https://github.com/ampproject/amphtml/releases/tag/1810152207300. This is currently in pre-release, which I think is tracked by https://github.com/ampproject/amphtml/issues/18649. Since it's this week's canary, it will probably make it to production the following Tuesday. cc @erwinmombay who can probably say for certain. 

Another example of why we need https://github.com/ampproject/amphtml/issues/18551. ;)

**Pull request merged**